### DISCUSSION TITLE
Do you plan to synchronize to theia 1.50.1

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#1356] [Q&A] [answered]
Hi,
    When updating theia to 1.50.1, 'Saveable' adds an onContentChanged event.  the 'GLSPSaveble' in theia-integration not implement it
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/9de0c404-edd6-4a62-af70-39d165bd2e85)
so open GLSPDiagramWidget always gets an error
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/c1638700-7d11-4a62-b50e-45aa1505dfe5)

**Comment by martin-fleck-at** [comment]
@wss29 Very good point! We'll look into it.

**Reply by martin-fleck-at** [reply]
Tobias opened a PR that will tackle that: https://github.com/eclipse-glsp/glsp-theia-integration/pull/220

**Comment by tortmayr** [comment, marked as answer]
@wss29 https://github.com/eclipse-glsp/glsp-theia-integration/pull/220 has been merged into master and we released a hotfix version for 2.1.1 that is compatible with Theia >= 1.49.0
https://www.npmjs.com/package/@eclipse-glsp/theia-integration/v/2.1.1-theia1.49.0

**Reply by wss29** [reply]
Hi @tortmayr
Could I use all version 2.2.0? Include glsp-client, glsp-node-server, glsp-theia-integration, it seems that their many versions between this three library,  and it is difficult to coordinate the dependency between them. and I have seen glsp-client release 3.0.0, does glsp-node-server, glsp-theia-integration plan to release 3.0.0?

**Reply by tortmayr** [reply]
As of now the latest release is 2.1.1 or (2.1.1-theia<version> for hotfix releases)

We were planning to release 2.2.0 last week but due to a bug in a our release pipeline we ended up with publishing incorrect packages and had to revert the pending release. All npm `2.2.0` npm version have been deprecated and should not be used.

We are working on a new release (2.2.1) which should be available by the end of the week. But for now the latest release is still 2.2.1

>and I have seen glsp-client release 3.0.0, does glsp-node-server, glsp-theia-integration plan to release 3.0.0?

The 3.0.0 tag has been pushed on accident and is remove now. For the time being GLSP will remain on 2.x versions, there are no plans to release a new major version yet.

**Reply by wss29** [reply]
Hi,
For 3.0.0, I have just seen this morning, but it seem be delete just now
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/b7226546-416f-427e-b6d5-15757ddb2fcc)

**Reply by tortmayr** [reply]
Ah yes. The 3.0.0 tag has been pushed on accident and is remove now.
